src/compose/ComposeBox.js

```diff
@@ -142,281 +87,415 @@ function randomInt(min, max) {
   return Math.floor(Math.random() * (max - min + 1)) + min;
 }
 
-const updateTextInput = (textInput, text) => {
-  if (textInput === null) {
-    // Depending on the lifecycle events this function is called from,
-    // this might not be set yet.
-    return;
```

Comment:
I like that this goes away!

Reply:
The conditional remains, for the reason described in the comment, but the comment is removed. Possibly the comment should remain? 🤔

Note the optional chaining:

```
messageInputRef.current?.setNativeProps({ text: newValue });
```

Reply:
Hmm, I see -- yeah, I'd missed that `?`.

I was imagining that it'd become unnecessary because the concept of "lifecycle events" isn't the same anymore now that it's a function component. But I guess even if that's true, the types don't know that -- the value does start out as null, and even if there is some argument that proves that these functions never get called before it's set to non-null (and stays that way), that reasoning is beyond the type system's ken. So there needs to be a conditional anyway.

I don't think the comment needs to stay. We have other references to `messageInputRef.current?.foo` -- it's essentially the only way to use it. Checking the history, that comment was introduced in cbd180dbbb5f224b59fccde7d576555ce00ceaed back in 2018:
```diff
+export const updateTextInput = (textInput: TextInput, text: string): void => {
+  if (!textInput) {
+    // Depending on the lifecycle events this function is called from,
+    // this might not be set yet.
+    return;
```

where in fact, if you look closely, it's *contradicting* what the type signature says. A few months later in 711e7867990bd2b41c26f952ef9a1341cb4cf687 we fixed the type:
```
    compose box: Fix a nullability nit in type signature.
    
    The logic here is already correct, with a condition just under this;
    just need to make the signature agree.
    
    This issue is found by Flow when we unify the ComposeBox files,
    coming soon -- Flow unfortunately doesn't check *.android.js.

diff --git a/src/compose/ComposeBox.android.js b/src/compose/ComposeBox.android.
js
--- a/src/compose/ComposeBox.android.js
+++ b/src/compose/ComposeBox.android.js
@@ -77,16 +77,16 @@
-export const updateTextInput = (textInput: TextInput, text: string): void => {
+export const updateTextInput = (textInput: ?TextInput, text: string): void => {
```

but I didn't go and update the comment then.